.travis.yml

```diff
@@ -27,6 +27,7 @@ before_script:
   - node_modules/.bin/bower install
   - node_modules/.bin/grunt build
   - python manage.py print_settings
+  - python manage.py collectstatic_js_reverse
```

Comment:
I guess this is fine as long as we are relying on global libraries and can't get the urls via an API call for example. Something like that is required to really make the frontend not dependent on the backend.

However, I would prefer if developers didn't have to remember always to run this command every time Django urls change. Could you copy this generated file to `socialhome/streams/app/tests/fixtures` and require it from there? Also the dev docs should have a mention for developers to pre-generate and copy this file every time they change backend url config (relating to API's). Basically [here](https://socialhome.readthedocs.io/en/latest/development.html) somewhere, maybe a new section after "Linters"?

Otherwise every developer needs to remember to run this command every single time someone changes the url config. This is going to be bad when we have 100 developers :)

This travis call can then be removed too. How does this sound?

Reply:
I'm ok with that.

Reply:
Here is my diff that seems to work, without doing a collect js urls:

    diff --git a/mocha/setup.js b/mocha/setup.js
    index ae05b53..4b9cbad 100644
    --- a/mocha/setup.js
    +++ b/mocha/setup.js
    @@ -35,7 +35,7 @@ global.WebSocket.prototype.close = function () {} // eslint-disable-line func-na
     global.Sinon = require("sinon").sandbox.create()
     
     // Trap calls to global library `Urls`
    -const file = "staticfiles/django_js_reverse/js/reverse"
    +const file = "socialhome/streams/app/tests/fixtures/Url"
     
     /* eslint-disable global-require, import/no-dynamic-require */
     try {
    diff --git a/socialhome/streams/app/tests/fixtures/Url.js b/socialhome/streams/app/tests/fixtures/Url.js
    new file mode 100644
    index 0000000..e132cfd
    --- /dev/null
    +++ b/socialhome/streams/app/tests/fixtures/Url.js
    @@ -0,0 +1,13 @@
    +this.Urls=(function(){var Urls={};var self={url_patterns:{}};var _get_url=function(url_pattern){return function(){var _arguments,index,url,url_arg,url_args,_i,_len,_ref,_ref_list,match_ref,provided_keys,build_kwargs;_arguments=arguments;_ref_list=self.url_patterns[url_pattern];if(arguments.length==1&&typeof(arguments[0])=="object"){var provided_keys_list=Object.keys(arguments[0]);provided_keys={};for(_i=0;_i<provided_keys_list.length;_i++)
    +provided_keys[provided_keys_list[_i]]=1;match_ref=function(ref)
    +{var _i;if(ref[1].length!=provided_keys_list.length)
    +return false;for(_i=0;_i<ref[1].length&&ref[1][_i]in provided_keys;_i++);return _i==ref[1].length;}
    +build_kwargs=function(keys){return _arguments[0];}}else{match_ref=function(ref)
    +{return ref[1].length==_arguments.length;}
    +build_kwargs=function(keys){var kwargs={};for(var i=0;i<keys.length;i++){kwargs[keys[i]]=_arguments[i];}
    +return kwargs;}}
    +for(_i=0;_i<_ref_list.length&&!match_ref(_ref_list[_i]);_i++);if(_i==_ref_list.length)
    +return null;_ref=_ref_list[_i];url=_ref[0],url_args=build_kwargs(_ref[1]);for(url_arg in url_args){var url_arg_value=url_args[url_arg];if(url_arg_value===undefined||url_arg_value===null){url_arg_value='';}else{url_arg_value=url_arg_value.toString();}
    +url=url.replace("%("+url_arg+")s",url_arg_value);}
    +return'/'+url;};};var name,pattern,url,url_patterns,_i,_len,_ref;url_patterns=[['account_change_password',[['accounts/password/change/',[]]],],['account_confirm_email',[['accounts/confirm\u002Demail/%(key)s/',['key',]]],],['account_email',[['accounts/email/',[]]],],['account_email_verification_sent',[['accounts/confirm\u002Demail/',[]]],],['account_inactive',[['accounts/inactive/',[]]],],['account_login',[['accounts/login/',[]]],],['account_logout',[['accounts/logout/',[]]],],['account_reset_password',[['accounts/password/reset/',[]]],],['account_reset_password_done',[['accounts/password/reset/done/',[]]],],['account_reset_password_from_key',[['accounts/password/reset/key/%(uidb36)s\u002D%(key)s/',['uidb36','key',]]],],['account_reset_password_from_key_done',[['accounts/password/reset/key/done/',[]]],],['account_set_password',[['accounts/password/set/',[]]],],['account_signup',[['accounts/signup/',[]]],],['admin:account_emailaddress_add',[['admin/account/emailaddress/add/',[]]],],['admin:account_emailaddress_change',[['admin/account/emailaddress/%(_0)s/change/',['_0',]]],],['admin:account_emailaddress_changelist',[['admin/account/emailaddress/',[]]],],['admin:account_emailaddress_delete',[['admin/account/emailaddress/%(_0)s/delete/',['_0',]]],],['admin:account_emailaddress_history',[['admin/account/emailaddress/%(_0)s/history/',['_0',]]],],['admin:app_list',[['admin/%(app_label)s/',['app_label',]]],],['admin:auth_group_add',[['admin/auth/group/add/',[]]],],['admin:auth_group_change',[['admin/auth/group/%(_0)s/change/',['_0',]]],],['admin:auth_group_changelist',[['admin/auth/group/',[]]],],['admin:auth_group_delete',[['admin/auth/group/%(_0)s/delete/',['_0',]]],],['admin:auth_group_history',[['admin/auth/group/%(_0)s/history/',['_0',]]],],['admin:auth_user_password_change',[['admin/users/user/%(_0)s/password/',['_0',]]],],['admin:authtoken_token_add',[['admin/authtoken/token/add/',[]]],],['admin:authtoken_token_change',[['admin/authtoken/token/%(_0)s/change/',['_0',]]],],['admin:authtoken_token_changelist',[['admin/authtoken/token/',[]]],],['admin:authtoken_token_delete',[['admin/authtoken/token/%(_0)s/delete/',['_0',]]],],['admin:authtoken_token_history',[['admin/authtoken/token/%(_0)s/history/',['_0',]]],],['admin:dynamic_preferences_globalpreferencemodel_add',[['admin/dynamic_preferences/globalpreferencemodel/add/',[]]],],['admin:dynamic_preferences_globalpreferencemodel_change',[['admin/dynamic_preferences/globalpreferencemodel/%(_0)s/change/',['_0',]]],],['admin:dynamic_preferences_globalpreferencemodel_changelist',[['admin/dynamic_preferences/globalpreferencemodel/',[]]],],['admin:dynamic_preferences_globalpreferencemodel_delete',[['admin/dynamic_preferences/globalpreferencemodel/%(_0)s/delete/',['_0',]]],],['admin:dynamic_preferences_globalpreferencemodel_history',[['admin/dynamic_preferences/globalpreferencemodel/%(_0)s/history/',['_0',]]],],['admin:dynamic_preferences_users_userpreferencemodel_add',[['admin/dynamic_preferences_users/userpreferencemodel/add/',[]]],],['admin:dynamic_preferences_users_userpreferencemodel_change',[['admin/dynamic_preferences_users/userpreferencemodel/%(_0)s/change/',['_0',]]],],['admin:dynamic_preferences_users_userpreferencemodel_changelist',[['admin/dynamic_preferences_users/userpreferencemodel/',[]]],],['admin:dynamic_preferences_users_userpreferencemodel_delete',[['admin/dynamic_preferences_users/userpreferencemodel/%(_0)s/delete/',['_0',]]],],['admin:dynamic_preferences_users_userpreferencemodel_history',[['admin/dynamic_preferences_users/userpreferencemodel/%(_0)s/history/',['_0',]]],],['admin:index',[['admin/',[]]],],['admin:jsi18n',[['admin/jsi18n/',[]]],],['admin:login',[['admin/login/',[]]],],['admin:logout',[['admin/logout/',[]]],],['admin:password_change',[['admin/password_change/',[]]],],['admin:password_change_done',[['admin/password_change/done/',[]]],],['admin:sites_site_add',[['admin/sites/site/add/',[]]],],['admin:sites_site_change',[['admin/sites/site/%(_0)s/change/',['_0',]]],],['admin:sites_site_changelist',[['admin/sites/site/',[]]],],['admin:sites_site_delete',[['admin/sites/site/%(_0)s/delete/',['_0',]]],],['admin:sites_site_history',[['admin/sites/site/%(_0)s/history/',['_0',]]],],['admin:socialaccount_socialaccount_add',[['admin/socialaccount/socialaccount/add/',[]]],],['admin:socialaccount_socialaccount_change',[['admin/socialaccount/socialaccount/%(_0)s/change/',['_0',]]],],['admin:socialaccount_socialaccount_changelist',[['admin/socialaccount/socialaccount/',[]]],],['admin:socialaccount_socialaccount_delete',[['admin/socialaccount/socialaccount/%(_0)s/delete/',['_0',]]],],['admin:socialaccount_socialaccount_history',[['admin/socialaccount/socialaccount/%(_0)s/history/',['_0',]]],],['admin:socialaccount_socialapp_add',[['admin/socialaccount/socialapp/add/',[]]],],['admin:socialaccount_socialapp_change',[['admin/socialaccount/socialapp/%(_0)s/change/',['_0',]]],],['admin:socialaccount_socialapp_changelist',[['admin/socialaccount/socialapp/',[]]],],['admin:socialaccount_socialapp_delete',[['admin/socialaccount/socialapp/%(_0)s/delete/',['_0',]]],],['admin:socialaccount_socialapp_history',[['admin/socialaccount/socialapp/%(_0)s/history/',['_0',]]],],['admin:socialaccount_socialtoken_add',[['admin/socialaccount/socialtoken/add/',[]]],],['admin:socialaccount_socialtoken_change',[['admin/socialaccount/socialtoken/%(_0)s/change/',['_0',]]],],['admin:socialaccount_socialtoken_changelist',[['admin/socialaccount/socialtoken/',[]]],],['admin:socialaccount_socialtoken_delete',[['admin/socialaccount/socialtoken/%(_0)s/delete/',['_0',]]],],['admin:socialaccount_socialtoken_history',[['admin/socialaccount/socialtoken/%(_0)s/history/',['_0',]]],],['admin:users_user_add',[['admin/users/user/add/',[]]],],['admin:users_user_change',[['admin/users/user/%(_0)s/change/',['_0',]]],],['admin:users_user_changelist',[['admin/users/user/',[]]],],['admin:users_user_delete',[['admin/users/user/%(_0)s/delete/',['_0',]]],],['admin:users_user_history',[['admin/users/user/%(_0)s/history/',['_0',]]],],['admin:view_on_site',[['admin/r/%(content_type_id)s/%(object_id)s/',['content_type_id','object_id',]]],],['api\u002Ddocs',[['api/',[]]],],['api\u002Dimage\u002Dupload',[['api/image\u002Dupload/',[]]],],['api\u002Dstreams:followed',[['api/streams/followed/',[]]],],['api\u002Dstreams:public',[['api/streams/public/',[]]],],['api\u002Dstreams:tag',[['api/streams/tag/%(name)s/',['name',]]],],['api:api\u002Droot',[['api/.%(format)s',['format',],],['api/',[]]],],['api:content\u002Ddetail',[['api/content/%(pk)s.%(format)s',['pk','format',],],['api/content/%(pk)s/',['pk',]]],],['api:content\u002Dlist',[['api/content.%(format)s',['format',],],['api/content/',[]]],],['api:content\u002Dreplies',[['api/content/%(pk)s/replies.%(format)s',['pk','format',],],['api/content/%(pk)s/replies/',['pk',]]],],['api:content\u002Dshare',[['api/content/%(pk)s/share.%(format)s',['pk','format',],],['api/content/%(pk)s/share/',['pk',]]],],['api:profile\u002Dadd\u002Dfollower',[['api/profiles/%(pk)s/add_follower.%(format)s',['pk','format',],],['api/profiles/%(pk)s/add_follower/',['pk',]]],],['api:profile\u002Ddetail',[['api/profiles/%(pk)s.%(format)s',['pk','format',],],['api/profiles/%(pk)s/',['pk',]]],],['api:profile\u002Dremove\u002Dfollower',[['api/profiles/%(pk)s/remove_follower.%(format)s',['pk','format',],],['api/profiles/%(pk)s/remove_follower/',['pk',]]],],['api:user\u002Ddetail',[['api/users/%(pk)s.%(format)s',['pk','format',],],['api/users/%(pk)s/',['pk',]]],],['bookmarklet',[['bookmarklet/',[]]],],['content:bookmarklet',[['content/bookmarklet/',[]]],],['content:create',[['content/create/',[]]],],['content:delete',[['content/%(pk)s/~delete/',['pk',]]],],['content:reply',[['content/%(pk)s/~reply/',['pk',]]],],['content:update',[['content/%(pk)s/~edit/',['pk',]]],],['content:view',[['content/%(pk)s/',['pk',]]],],['content:view\u002Dby\u002Dguid',[['content/%(guid)s/',['guid',]]],],['content:view\u002Dby\u002Dslug',[['content/%(pk)s/%(slug)s/',['pk','slug',]]],],['djdt:debug\u002Duserpanel\u002Dlogin',[['__debug__/users/login/%(pk)s',['pk',]]],],['djdt:debug\u002Duserpanel\u002Dlogin\u002Dform',[['__debug__/users/login/',[]]],],['djdt:debug\u002Duserpanel\u002Dlogout',[['__debug__/users/logout',[]]],],['djdt:render_panel',[['__debug__/render_panel/',[]]],],['djdt:sql_explain',[['__debug__/sql_explain/',[]]],],['djdt:sql_profile',[['__debug__/sql_profile/',[]]],],['djdt:sql_select',[['__debug__/sql_select/',[]]],],['djdt:template_source',[['__debug__/template_source/',[]]],],['dynamic_preferences.global',[['preferences/global/',[]]],],['dynamic_preferences.global.section',[['preferences/global/%(section)s',['section',]]],],['dynamic_preferences.user',[['preferences/user/',[]]],],['dynamic_preferences.user.section',[['preferences/user/%(section)s',['section',]]],],['federate:content\u002Dfetch',[['fetch/%(objtype)s/%(guid)s',['objtype','guid',]]],],['federate:content\u002Dxml',[['p/%(guid)s.xml',['guid',]]],],['federate:hcard',[['hcard/users/%(guid)s',['guid',]]],],['federate:host\u002Dmeta',[['.well\u002Dknown/host\u002Dmeta',[]]],],['federate:nodeinfo',[['nodeinfo/1.0',[]]],],['federate:nodeinfo\u002Dwellknown',[['.well\u002Dknown/nodeinfo',[]]],],['federate:receive\u002Dpublic',[['receive/public/',[]]],],['federate:receive\u002Duser',[['receive/users/%(guid)s/',['guid',]]],],['federate:social\u002Drelay',[['.well\u002Dknown/x\u002Dsocial\u002Drelay',[]]],],['federate:webfinger',[['webfinger',[]]],],['home',[['',[]]],],['javascript\u002Dcatalog',[['jsi18n/',[]]],],['js_reverse',[['jsreverse/',[]]],],['markdownx_markdownify',[['markdownx/markdownify/',[]]],],['markdownx_upload',[['markdownx/upload/',[],],['markdownx/upload/',[]]],],['mocha_view',[['mocha/%(app)s/%(config)s',['app','config',],],['mocha/%(app)s/',['app',]]],],['rest_framework:login',[['api\u002Dauth/login/',[]]],],['rest_framework:logout',[['api\u002Dauth/logout/',[]]],],['rq_actions',[['django\u002Drq/queues/actions/%(queue_index)s/',['queue_index',]]],],['rq_clear',[['django\u002Drq/queues/%(queue_index)s/empty/',['queue_index',]]],],['rq_deferred_jobs',[['django\u002Drq/queues/%(queue_index)s/deferred/',['queue_index',]]],],['rq_delete_job',[['django\u002Drq/queues/%(queue_index)s/%(job_id)s/delete/',['queue_index','job_id',]]],],['rq_finished_jobs',[['django\u002Drq/queues/%(queue_index)s/finished/',['queue_index',]]],],['rq_home',[['django\u002Drq/',[]]],],['rq_job_detail',[['django\u002Drq/queues/%(queue_index)s/%(job_id)s/',['queue_index','job_id',]]],],['rq_jobs',[['django\u002Drq/queues/%(queue_index)s/',['queue_index',]]],],['rq_requeue_all',[['django\u002Drq/queues/%(queue_index)s/requeue\u002Dall/',['queue_index',]]],],['rq_requeue_job',[['django\u002Drq/queues/%(queue_index)s/%(job_id)s/requeue/',['queue_index','job_id',]]],],['rq_started_jobs',[['django\u002Drq/queues/%(queue_index)s/started/',['queue_index',]]],],['search:global',[['search/',[]]],],['socialaccount_connections',[['accounts/social/connections/',[]]],],['socialaccount_login_cancelled',[['accounts/social/login/cancelled/',[]]],],['socialaccount_login_error',[['accounts/social/login/error/',[]]],],['socialaccount_signup',[['accounts/social/signup/',[]]],],['streams\u002Dlegacy:followed',[['followed/',[]]],],['streams\u002Dlegacy:public',[['public/',[]]],],['streams\u002Dlegacy:tag',[['tags/%(name)s/',['name',]]],],['streams:followed',[['streams/followed/',[]]],],['streams:public',[['streams/public/',[]]],],['streams:tag',[['streams/tag/%(name)s/',['name',]]],],['users:all\u002Dcontent',[['u/%(username)s/all/',['username',]]],],['users:api\u002Dtoken',[['u/~api\u002Dtoken/',[]]],],['users:contacts\u002Dfollowed',[['contacts/',[]]],],['users:detail',[['u/%(username)s/',['username',]]],],['users:list',[['u/',[]]],],['users:picture\u002Dupdate',[['u/~update\u002Dpicture/',[]]],],['users:profile\u002Dall\u002Dcontent',[['p/%(guid)s/all/',['guid',]]],],['users:profile\u002Ddetail',[['p/%(guid)s/',['guid',]]],],['users:profile\u002Dorganize',[['p/~organize/',[]]],],['users:profile\u002Dupdate',[['p/~update/',[]]]]];self.url_patterns={};for(_i=0,_len=url_patterns.length;_i<_len;_i++){_ref=url_patterns[_i],name=_ref[0],pattern=_ref[1];self.url_patterns[name]=pattern;url=_get_url(name);Urls[name]=url;Urls[name.replace(/-/g,'_')]=url;}
    +return Urls;})();
    \ No newline at end of file